lib/inspec/cli.rb

```diff
@@ -305,7 +305,9 @@ def detect
       puts res.to_json
     else
       ui.headline("Platform Details")
-      ui.plain Inspec::BaseCLI.format_platform_info(params: res, indent: 0, color: 36)
+
+      detect_cmd_ui_color = ui.color? ? 36 : 0
+      ui.plain Inspec::BaseCLI.format_platform_info(params: res, indent: 0, color: detect_cmd_ui_color)
```

Comment:
I think `Inspec::BaseCLI.format_platform_info` needs to be changed to avoid outputting color codes entirely when color == 0.